### PR TITLE
perf(nns): Add benchmarks for periodic tasks

### PR DESCRIPTION
**Pull request by jasonz-dfinity**
# Why

There was a recent regression on how we list neurons ready to unstake maturity or spawn. Adding benchmarks for those will help prevent them in the future (although, currently they are not enabled in CI yet).

# What

Adding benchmarks for `list_ready_to_spawn_neuron_ids` and `list_neurons_ready_to_unstake_maturity`. Additional dimensions are: (1) whether the "active neuron in stable memory" is enabled (2) whether there are actually meaningful return value.

# Testing

When running against changes in https://github.com/dfinity/ic/pull/2726 , we get the following results:

---------------------------------------------------
Benchmark: list_ready_to_spawn_neuron_ids_heap
  total:
    instructions: 455.22 K (improved by 86.76%)
    heap_increase: 0 pages (no change)
    stable_memory_increase: 0 pages (no change)

---------------------------------------------------

Benchmark: list_ready_to_spawn_neuron_ids_stable
  total:
    instructions: 710.56 M (0.00%) (change within noise threshold)
    heap_increase: 0 pages (no change)
    stable_memory_increase: 0 pages (no change)

---------------------------------------------------

Benchmark: list_neurons_ready_to_unstake_maturity_heap
  total:
    instructions: 463.15 K (improved by 86.54%)
    heap_increase: 0 pages (no change)
    stable_memory_increase: 0 pages (no change)

---------------------------------------------------

Benchmark: list_neurons_ready_to_unstake_maturity_stable
  total:
    instructions: 710.57 M (0.00%) (change within noise threshold)
    heap_increase: 0 pages (no change)
    stable_memory_increase: 0 pages (no change)

---------------------------------------------------

### CODE REVIEW

**Comment by daniel-wong-dfinity-org** [comment]
Thank you for measuring!

Not sure how to interpret the data in the PR description. I can see that in some cases, there are dramatic improvements (and in other cases, the improvement is marginal). What I'm not sure is, how does this translate into "responsiveness" experienced by users (not necessarily consciously noticed, but just experienced). For example, I assume that even if we make these functions use 95% less time, I doubt that users get responses in 95% less time. I guess these functions are used in heartbeat -> it is hard to say how this affects responsiveness.

In short, my question is "What does this table mean?". I guess one thing I can say for sure is the the fix in 2726 does not HURT anything...

Ultimately, the question is, "Is expediting 2726 worth it?". With this data, I'm still not really sure how to answer that. Thus, I remain on the fence whether expediting is a good idea. FWIW, [Sam Burri has given us a green light to expedite][green light] (if we want).

https://dfinity.slack.com/archives/C081YMWE3L4/p1732198360213759

**Comment by jasonz-dfinity** [comment]
> Thank you for measuring!
> 
> Not sure how to interpret the data in the PR description. I can see that in some cases, there are dramatic improvements (and in other cases, the improvement is marginal). What I'm not sure is, how does this translate into "responsiveness" experienced by users (not necessarily consciously noticed, but just experienced). For example, I assume that even if we make these functions use 95% less time, I doubt that users get responses in 95% less time. I guess these functions are used in heartbeat -> it is hard to say how this affects responsiveness.
> 
> In short, my question is "What does this table mean?". I guess one thing I can say for sure is the the fix in 2726 does not HURT anything...
> 
> Ultimately, the question is, "Is expediting 2726 worth it?". With this data, I'm still not really sure how to answer that. Thus, I remain on the fence whether expediting is a good idea. FWIW, [Sam Burri has given us a green light to expedite][green light] (if we want).
> 
> https://dfinity.slack.com/archives/C081YMWE3L4/p1732198360213759

Thanks. I didn't mean for the benchmark to be the justification, simply trying to put it in place so that it can be avoided in the future.
